TicTacToe/TicTacToe.js

```diff
@@ -0,0 +1,157 @@
+class Game {
+    moves = 0;
+    user = 'X';
+    board = [['-', '-', '-'], ['-', '-', '-'], ['-', '-', '-']];
+    messages = "Game - START!";
+    complete = false;
+
+    reset() {
+        this.moves = 0;
+        this.user = 'X';
+        this.board = [['-', '-', '-'], ['-', '-', '-'], ['-', '-', '-']];
+        this.messages = "Game - START!";
+        this.complete = false;
+    }
+
+    getMoves() {
+        return this.moves;
+    };
+
+    displayBoard() {
+        let text = '';
+        for (let i = 0; i < 3; i++) {
+            for (let k = 0; k < 3; k++) {
+                text += " " + this.board[i][k];
+            }
+            text += "\n";
+        }
+        console.log(text);
+        return text;
+    };
+
+    getMessages() {
+        return this.messages;
+    }
+
+    makeMove(x, y) {
+        this.messages = "";
+        if (this.complete) {
+            this.messages = "GAME COMPLETED";
+            return;
+        }
+
+        if (this.board[x][y] != '-') {
+            this.messages = "Invalid move! Try again!";
+            if (this.user == 'X') {
+                return 'O';
+            } else {
+                return 'X';
+            }
+        }
+
+        this.board[x][y] = this.user;
+        this.moves++;
+
+        if (this.isWin(x, y)) {
+            this.messages = this.user + " is the winner!";
+            this.complete = true;
+            return this.user;
+        }
+
+        if (this.moves == 9) {
+            this.messages = "It's a tie!";
+            return this.user;
+        }
+
+        if (this.user == 'X') {
+            this.user = 'O';
+            return 'X';
+        } else {
+            this.user = 'X';
+            return 'O';
+        }
+    };
+
+    isWin(x, y) {
+        let win1 = true;
+        let win2 = true;
+        let win3 = true;
+        let win4 = true;
+
+        // check row
+        for (let i = 0; i < 3; i++) {
```

Comment:
Best not to hardcode `3` or `9` -- better to declare a board size constant `N` and perform calculations off of it.